src/library/scala/collection/mutable/HashTable.scala

```diff
@@ -363,11 +363,11 @@ trait HashTable[A, Entry >: Null <: HashEntry[A, Entry]] extends HashTable.HashU
   // Note:
   // we take the most significant bits of the hashcode, not the lower ones
   // this is of crucial importance when populating the table in parallel
+  // See: https://graphics.stanford.edu/~seander/bithacks.html#IntegerLogDeBruijn for explanation and BitManipulationBenchmark for alternatives
+  private val exponents = Array(32, 31, 4, 30, 3, 18, 8, 29, 2, 10, 12, 17, 7, 15, 28, 24, 1, 5, 19, 9, 11, 13, 16, 25, 6, 20, 14, 26, 21, 27, 22, 23)
   protected final def index(hcode: Int) = {
-    val ones = table.length - 1
-    val improved = improve(hcode, seedvalue)
-    val shifted = (improved >> (32 - java.lang.Integer.bitCount(ones))) & ones
-    shifted
+    val exponent = exponents((table.length * 0x077CB531) >>> 27)
+    (improve(hcode, seedvalue) >>> exponent) & (table.length - 1)
```

Comment:
> assume that table.length is an exact power of 2

Indeed, https://github.com/scala/scala/pull/5537/commits/d6778238fe91d8dbb9d13a42afb5113ed6c96526#diff-a24c1e91e1a44ff65f3da12201465cebR154 resizes this to be a power of two always (please verify).

> numberOfTrailingZeros(table.length). Have you tried that? 

Yes, please check out the benchmarks below:
https://github.com/scala/scala/pull/5537/commits/d6778238fe91d8dbb9d13a42afb5113ed6c96526#diff-9ee4c1bdb2df6f45b8c03d48dd2b0b0dR41